src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            int newCapacity = (int)(elements.length * 1.5);
+            elements = Arrays.copyOf(elements, newCapacity);
+        }
+    }
+
+    private void ensureCapacity(int DEFAULT_CAPACITY) {
```

Comment:
Overloading the ensureCapacity method with a parameter that has the same name as a constant field (DEFAULT_CAPACITY) is confusing. Consider using a different parameter name to avoid shadowing the constant field.